content: simulation update (#165)
fictionlab/docs: docs/integrations/software/gazebo-simulation.mdx

Reviewer: This link is incorrect - it leads to ros2 control installation and not to ros2 installation. This should link to: [https://docs.ros.org/en/jazzy/Installation.html](https://docs.ros.org/en/jazzy/Installation.html)

Author: Link changed

Reviewer: ```suggestion
namespace for the argument `cmd_vel_out`. In the example it's `leo_sim1`. If the
`cmd_vel_out` argument is not provided, the node will use the default namespace and topic - `/cmd_vel`.
```

description: >-
  Step-by-step guide to set up and run the simulation environment in Gazebo with
  ROS.
image: /img/robots/leo/integrations/gazebo-zed2/3-leo-rovers-gazebo.webp
---

# Gazebo Simulation for Mobile Robots

In this tutorial you will learn how to launch a Gazebo simulation with the Leo
Rover robot model.

## What to expect?

After completing this guide, you will have a fully simulated Leo Rover running
in Gazebo that behaves just like the real one. The simulated rover publishes
camera images, IMU readings and odometry, and can be driven either by publishing
velocity commands or with a joystick. You can spawn multiple rovers and choose
from several worlds.

The following topics are available after running the simulation (namespace
`robot_ns:=leo_sim1`):

```
/clock
/leo_sim1/camera/camera_info
/leo_sim1/camera/image_raw
/leo_sim1/camera/image_raw/compressed
/leo_sim1/camera/image_raw/compressedDepth
/leo_sim1/camera/image_raw/theora
/leo_sim1/camera/image_raw/zstd
/leo_sim1/cmd_vel
/leo_sim1/imu/data_raw
/leo_sim1/joint_states
/leo_sim1/odom
/leo_sim1/robot_description
/leo_sim1/tf
/tf

> /tf_static

Reviewer: List the default topic list for a robot without namespace

Author: Done

Alternatively, [here](https://github.com/LeoRover/leo_simulator-ros2/tree/jazzy)

> are instructions how to build the packages from source.

Reviewer: ```suggestion
Alternatively, you can build the packages from source by following the instructions [here](https://github.com/LeoRover/leo_simulator-ros2/tree/jazzy).
```

refer to
[leo_gz_worlds](https://github.com/LeoRover/leo_simulator-ros2/tree/jazzy/leo_gz_worlds)
package). For example you can launch simulation of an ERC 2021 competition

> terrain and put the robot inside some namespace like this:

Reviewer: ```suggestion
terrain and put the robot inside a namespace like this:
```
